tensorflow/python/ops/losses/losses_impl.py

```diff
@@ -419,6 +420,9 @@ def huber_loss(labels, predictions, weights=1.0, delta=1.0, scope=None,
     raise ValueError("predictions must not be None.")
   with ops.name_scope(scope, "huber_loss",
                       (predictions, labels, weights)) as scope:
+    if not gen_math_ops.is_finite(delta):
```

Comment:
This uses the result of a tensor operation as a python boolean, which is only supported inside tf.function / autograph.

As it is right now I believe this code always raises an exception, so please add unit tests to convince yourself the implementation is correct.

Also, why not just modify the huber loss kernel to do the right thing in the presence of infinity there?

Reply:
Great point! Acknowledged.